components/clarity-repl/src/analysis/ast_dependency_detector.rs

```diff
@@ -667,6 +667,20 @@ impl<'a> ASTVisitor<'a> for ASTDependencyDetector<'a> {
         );
         true
     }
+
+    fn visit_define_constant(
+        &mut self,
+        expr: &'a SymbolicExpression,
+        name: &'a ClarityName,
+        value: &'a SymbolicExpression,
+    ) -> bool {
+        if let Some(Value::Principal(PrincipalData::Contract(contract_principal))) =
+            value.match_literal_value()
+        {
+            self.add_dependency(self.current_contract.unwrap(), contract_principal);
```

Comment:
This is technically overly cautious, as you could store the contract principal in a constant and never use it, or only use it as a principal, but not use it as the target of a `contract-call?`. If we're okay taking this overly cautious approach, then I think the whole dependency detector could be simplified to just an implementation of `visit_literal_value` that looks for a contract principal.

Reply:
Ah yes of course.
So I should check that the constant is effectively used in a `call-contract?`

Reply:
Yeah, and I just realized you might have to. I think I remember ALEX or Arkadiko having an unusual situation where contract A references B, but B calls A, so that would report a circular dependence if we tried the simplification.

Reply:
Ah you mean they kind of bypass the circular dependency detection by storing contract in constants. 
That might be an issue indeed. The repo I'm currently testing against is a new Alex set up

Reply:
Right, like they need to pass the contract principal as an argument, but not to actually call it. Something like that.

Reply:
Ah ok I see what you mean. 

My worry now is that we were probably missing circular dependencies (because we were missing dependencies). So some projects might break with the fix. 
But I guess it makes more sense

Reply:
If there were real circular dependencies, they wouldn't have been able to deploy any way, so I'd say it's fine if you break those.